mods/default/functions.lua

```diff
@@ -366,8 +364,7 @@ minetest.register_abm({
 		end
 
 		-- Look for likely neighbors.
-		local p2 = minetest.find_node_near(pos, 1, {"default:dirt_with_grass",
-				"default:dirt_with_dry_grass", "default:dirt_with_snow"})
+		local p2 = minetest.find_node_near(pos, 1, "group:spreads_on_dirt")
```

Comment:
Groups stand in for items, so this groups "anything, that is spreading on dirt", like flowers, grass, but not other dirt_with_grass imo.
And with flowers this might actually delete nodes. :-)

How about `grassy_dirt` or  alternatively just `dirt` with a node definition property check of `spreads_on_dirt` after?